commons-logs/src/main/java/eu/europeana/api/commons/logs/LoggableDispatcherServlet.java

```diff
@@ -0,0 +1,213 @@
+        } catch (Exception e) {
+            LOG.error("Exception occurred while retrieving HandlerExecutionChain from HttpServletRequest", e);
+        }
+
+        try {
+            super.doDispatch(request, response);
+        } catch (Exception e) {
+            LOG.error("Exception occurred while invoking DispatcherServlet", e);
+        } finally {
+            long processTime = System.currentTimeMillis() - startTime;
+            logRequest(request, response, handler, processTime, serverTime);
+            updateResponse(response);
+        }
+    }
+
+    /**
+     * Forms the LogMessage
+     *
+     * @param requestToCache
+     * @param responseToCache
+     * @param handler
+     */
+    private void logRequest(HttpServletRequest requestToCache, HttpServletResponse responseToCache, HandlerExecutionChain handler,
+                            long processTime, String serverTime) {
+        LogMessage logMessage = new LogMessage();
+
+        // TODO Have to decide if we will still set this values here
+        logMessage.setApp_guid(""); // app_id
+
+        logMessage.setApp_name(StringUtils.substringBefore(requestToCache.getHeader(LogConstants.HOST), LogConstants.HOST_SEPERATOR));
+        logMessage.setBytes(getResponsePayload(responseToCache).getBytes().length);
+        logMessage.setWskey(requestToCache.getParameter(LogConstants.WSKEY));
+        logMessage.setPort(requestToCache.getServerPort());
+        logMessage.setJavaMethod(handler.getHandler().toString());
+
+        // get the client's Ip and Geo Location
+        String clientIP = getClientIP(requestToCache);
+        // TODO Verify which Ip is this exactly; for now getting IPV4
+        String ipV4 = getIPV4();
+
+        logMessage.setClientIPv4(clientIP);
+        logMessage.setIpV4(ipV4);
+        logMessage.setClientLocation(getGeoLocation(clientIP));
+        logMessage.setLocation(getGeoLocation(ipV4));
+
+        // TODO figure out how we will calculate this
+        // Gorouter Time : is the total time it takes for the request to go through the
+        // Gorouter initially plus the time it takes for the response to travel back through the Gorouter. This does not include
+        // the time the request spends traversing the network to the app. This also does not include the time the
+        // app spends forming a response.
+        //logMessage.setGorouterTime(goRouterTime);
+
+        logMessage.setHttpVersion(StringUtils.substringAfter(requestToCache.getProtocol(), "/"));
+        logMessage.setReferer(requestToCache.getHeader(LogConstants.REFERER));
+        logMessage.setHttpStatus(responseToCache.getStatus());
+        logMessage.setProcessTime(processTime);
+        // server date and time
+        logMessage.setServerDate(StringUtils.substringBefore(serverTime, "T"));
+        logMessage.setServerTime(StringUtils.substringBetween(serverTime, "T", "Z"));
+        logMessage.setServerTimeZoneOffset(LogConstants.SERVER_TIMEZONE_OFFSET);
```

Comment:
The access logs generated by IBM have 1 date string, so I propose to do the same. It won't save much time, but it would avoid extra String parsing operations.

Reply:
yes, according to access log format, we can remove this